docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,209 @@
+  - Also, the namespaces catered by each NAC instance should be separate and not merge with each other
+  - We can facilitate this by accepting a list of NS that will be watched/catered by the particular NAC instance. We can do this during the installation time via DPA. There may be 2 ways which we will support of NS filtering:
+    - Direct List of NS Names `(Spec.features.nonAdmin.NSList[])`
+    - List of NS to filtered based on a LabelSelector `(Spec.features.nonAdmin.labelSelector)`
+    - By default, of none of the above are specified then NAC listens to all the Namespaces in the cluster
+    - We need to error out if NAC NS scopes overlap (i.e. NS watch list overlap)
+
+
+## Pre-requisites
+- **OADP installed**: OADP Operator must be installed and configured to use non-admin controller
+- **Non Admin Controller configured**: Data Protection Application (DPA) instance must configure Non Admin Controller to watch user namespace(s), by default it watches all Namespaces
+- **RBAC privileges for the user**: User must have the appropriate RBAC privileges to create Non Admin objects within the Namespace where Backup will be taken. An example of such ClusterRole, which may be added to the user with `RoleBinding`:
+    ```yaml
+    # permissions for end users to edit non-admin custom resources.
+    apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+    labels:
+        app.kubernetes.io/name: clusterrole
+        app.kubernetes.io/instance: nonadminbackup-editor-role
+        app.kubernetes.io/component: rbac
+        app.kubernetes.io/created-by: oadp-nac
+        app.kubernetes.io/part-of: oadp-nac
+        app.kubernetes.io/managed-by: kustomize
+    name: nonadminbackup-editor-role
+    rules:
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups
+        - nonadminrestores
+        - nonadminbackupstorageloations
+      verbs:
+        - create
+        - delete
+        - get
+        - list
+        - patch
+        - update
+        - watch
+    - apiGroups:
+        - nac.oadp.openshift.io
+      resources:
+        - nonadminbackups/status
+        - nonadminrestores/status
+        - nonadminbackupstoragelocations/status
+      verbs:
+        - get
+
+    ```
+- **Existence of Namespace:** The existence of namespace prior to performing non-admin backup/restore operations is a must.(for backup it will obviously exist but for restore it is an explicit hard precondition)
+## High-Level design
+
+### Components
+- **OADP Operator:** OADP is the OpenShift API for Data Protection operator. This open source operator sets up and installs Velero on the OpenShift platform, allowing users to backup and restore applications.
+- **Controllers:** The Non-Admin controller will pack the following controllers as part of it:
+    - **Non-Admin Backup (NAB) Controller:** The responsibilities of the NAB controller are:
```

Comment:
Should you add:
create Backup ?

Reply:
yeah `cascade any actions` should cover that